### PR TITLE
Change default level for deprecation logging

### PR DESCRIPTION
**Pull request by jakelandis**
This commit updates all deprecation message (except for REST
 compatible API messages) in 8.0+ to be emit at warning level.
Currently none of these have been removed in future versions (yet) so they 
should be logged at warning, not critical. 

This commit also changes the default `assertWarning` to assert at warning level
and introduces a new `assertCriticalWarning` to assert critical warnings.

### CODE REVIEW

**Review by pgomulka — APPROVED** [review]
LGTM

**Comment by jakelandis** [comment]
@pgomulka - can you take another look ? I also drop the non-settings down to warn level, such that nothing will currently log to critical.  It looks like like of changes, but the code change was just some creative use of Intellij's refactoring.

**Review by pgomulka — APPROVED** [review]
LGTM, I scrolled through the changes and all looks great

File: libs/x-content/src/test/java/org/elasticsearch/xcontent/ConstructingObjectParserTests.java [hunk 1]
```diff
@@ -817,7 +817,7 @@ public void testRemovalOfField() throws IOException {
             );
             StructRemovalField parse = StructRemovalField.PARSER.parse(parser, null);
 
-            assertWarnings("The field old_name has been removed and is being ignored");
+            assertCriticalWarnings("The field old_name has been removed and is being ignored");
```
Comment (jakelandis) [inline libs/x-content/src/test/java/org/elasticsearch/xcontent/ConstructingObjectParserTests.java:820]: REST API compatibility still warns at critical level.

File: server/src/main/java/org/elasticsearch/rest/DeprecationRestHandler.java [hunk 2]
```diff
@@ -65,7 +65,7 @@ public DeprecationRestHandler(
     @Override
     public void handleRequest(RestRequest request, RestChannel channel, NodeClient client) throws Exception {
         if (compatibleVersionWarning == false) {
-            deprecationLogger.critical(DeprecationCategory.API, deprecationKey, deprecationMessage);
+            deprecationLogger.warn(DeprecationCategory.API, deprecationKey, deprecationMessage);
```
Comment (jakelandis) [inline server/src/main/java/org/elasticsearch/rest/DeprecationRestHandler.java:68]: We will still need a way to emit critical level deprecations when decalaring a REST handler. I will log a follow up PR.

File: libs/x-content/src/test/java/org/elasticsearch/xcontent/ConstructingObjectParserTests.java (hunk 1 above)
Comment (pgomulka) [reply libs/x-content/src/test/java/org/elasticsearch/xcontent/ConstructingObjectParserTests.java:820]: I guess it is ok? compatibility warns about a feature that was already removed

File: x-pack/plugin/deprecation/qa/rest/src/javaRestTest/java/org/elasticsearch/xpack/deprecation/DeprecationHttpIT.java [hunk 3]
```diff
@@ -460,6 +460,58 @@ public void testDeprecationMessagesCanBeIndexed() throws Exception {
 
     }
 
+    /**
+     * Check that a deprecation message with CRITICAL level can be recorded to an index
+     */
+    public void testDeprecationCriticalWarnMessagesCanBeIndexed() throws Exception {
```
Comment (jakelandis) [inline x-pack/plugin/deprecation/qa/rest/src/javaRestTest/java/org/elasticsearch/xpack/deprecation/DeprecationHttpIT.java:466]: Now that the default is WARN, we needed to cover the case of a non-compat critical warning.

**Comment by jakelandis** [comment]
@elasticmachine update branch

**Comment by masseyke** [comment]
I believe all of your test failures are fixed by #80304 if we go that direction.

**Comment by jakelandis** [comment]
@elasticmachine update branch

**Comment by jakelandis** [comment]
@dakrone - can you take a look, this PR should be stable now. The initial (and primary change) was done with Intellij refactoring, however some the tests needed to updated to account for the new default warning level.

**Review by dakrone — APPROVED** [review]
LGTM

**Pull request merged**